infra/cmake/modules/ExternalBuildTools.cmake

```diff
@@ -54,7 +54,7 @@ function(ExternalBuild_CMake)
                             -G "${CMAKE_GENERATOR}"
                             -DCMAKE_INSTALL_PREFIX=${ARG_INSTALL_DIR}
                             -DCMAKE_BUILD_TYPE=Release
-                            -DCMAKE_CXX_FLAGS=${ARG_BUILD_FLAGS}
+                            -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} ${ARG_BUILD_FLAGS}
```

Comment:
plz split `compiler` and `infra` to separate PRs

Reply:
Q) is it OK if there is space in between?
`-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} ${ARG_BUILD_FLAGS}` or
`-DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} ${ARG_BUILD_FLAGS}"` ?